Update user guide to show the most updated information (#98)
AY2425S2-CS2103-F15-1/tp: docs/UserGuide.md

1. Ensure you have Java `17` or above installed in your Computer.<br>
   **Mac users:** Ensure you have the precise JDK version prescribed [here](https://se-education.org/guides/tutorials/javaInstallationMac.html).

1. Download the latest `.jar` file from [here](https://github.com/AY2425S2-CS2103-F15-1/tp/releases).

1. Copy the file to the folder you want to use as the _home folder_ for your TAskbook.

1. Open a command terminal,
   `cd` into the folder you put the jar file in,
   and use the `java -jar taskbook.jar` command to run the application.<br>
   A GUI similar to the below should appear in a few seconds.
   Note how the app contains some sample data.<br>
   ![Ui](images/Ui.png)

1. Type the command in the command box and press Enter to execute it.
   e.g. typing **`help`** and pressing Enter will open the help window.<br>
   Some example commands you can try:

   * `list` : Lists all contacts.

   * `add n/John Doe i/A0123456Z p/98765432 e/johnd@example.com h/@john_doe` :
     Adds a contact named `John Doe` to the Address Book.

>    * `delete 1` : Deletes the 3rd contact shown in the current list.

Reviewer: would this delete the 3rd contact shown in the current list?

### Adding a person: `add`

Adds a student to the address book.

Format: `add n/NAME i/STUDENT_ID p/PHONE e/EMAIL [t/TUTORIAL]...`

<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
A person can have any number of tags (including 0)
</div>

Examples:
* `add n/John Doe i/A0123456Z p/98765432 e/johnd@example.com h/@john_doe`
* `add n/John Doe i/A0123456Z p/98765432 e/johnd@example.com h/@john_doe t/CS2103_T01 t/CS2106_T02`

### Listing all students: `list`

Shows a list of all students in the address book.

Format: `list`

### Editing a person: `edit`

Edits an existing student in the address book.

Format: `edit INDEX [n/NAME] [i/STUDENT_ID] [p/PHONE] [e/EMAIL] [h/HANDLE] [t/TUTORIALS]...`

* Edits the person at the specified `INDEX`.
  The index refers to the index number shown in the displayed person list.
  The index **must be a positive integer** 1, 2, 3, ...
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.

> * When editing tutorial slots,
>   the existing tutorial slots of the person will be removed i.e adding of tutorials is not cumulative.

Reviewer 2: ```suggestion
* When editing tutorial slots,
  the existing set of tutorials allocated to the person will be replaced entirely with the new values provided i.e. adding of tutorials is not cumulative.
```

Format: `tutorial list`

### Adding a student to a tutorial: `tutorial add-student`

> Adds a student to a tutorial slot

Reviewer 3: ```suggestion
Adds a student to a tutorial slot.
```

Format: `tutorial add-student TUTORIAL_NAME s/STUDENT_INDEX...`

* `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, ...

Examples:

- `list` then `tutorial add-student cs2103-f15 s/1`:
  Adds the 1st student in the list to the tutorial slot `cs2103-f15`.

### Deleting a student to a tutorial: `tutorial delete-student`

> Deletes a student to a tutorial slot

Reviewer 2: ```suggestion
Deletes a student from a tutorial slot
```

> Deletes a student from a tutorial slot

Reviewer 3: ```suggestion
Deletes a student from a tutorial slot.
```

Format: `tutorial delete-student TUTORIAL_NAME s/STUDENT_INDEX...`

* `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, ...

Examples:

- `list` then `tutorial add-student cs2103-f15 s/1`:

>   Adds the 1st student in the list to the tutorial slot `cs2103-f15`.

Reviewer 2: Should change this to match ```tutorial delete-student```

## FAQ

**Q**: How do I transfer my data to another Computer?<br>
**A**: Install the app in the other computer
and overwrite the empty data file it creates with the file that contains the data of your previous TAskbook home folder.

--------------------------------------------------------------------------------------------------------------------

## Known issues

1. **When using multiple screens**, if you move the application to a secondary screen,
   and later switch to using only the primary screen,
   the GUI will open off-screen.
   The remedy is to delete the `preferences.json` file created by the application before running the application again.
2. **If you minimize the Help Window** and then run the `help` command
  (or use the `Help` menu, or the keyboard shortcut `F1`) again,
  the original Help Window will remain minimized,
  and no new Help Window will appear.
  The remedy is to manually restore the minimized Help Window.

--------------------------------------------------------------------------------------------------------------------

## Command summary

| **Action** | **Format, Examples** |
|------------|----------------------|
| **Help** | `help` |

> | **Add student** | `add n/NAME i/STUDENT_ID p/PHONE e/EMAIL [t/TUTORIAL]...`<br>e.g., `add n/John Doe i/A0123456Z p/98765432 e/johnd@example.com t/CS2103_T01 t/CS2106_T02` |

Reviewer 2: Are we including Telegram handle? i.e. ```t/@telegramhandle```

Author: We should add it, it's `h/HANDLE` yeah?
